### PR TITLE
[shopsys] changed used caches from Doctrine cache to Symfony cache

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| https://github.com/doctrine/cache is deprecated and should not be used anymore. Deprecated classes were stopped used in this PR to ensure a smooth upgrade of the library.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: project-base/config/packages/cache.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+framework:
+    cache:
+        pools:
+            main_friendly_url_slug_cache:
+                adapter: cache.adapter.redis
+                provider: snc_redis.main_friendly_url_slugs
+            shopsys.shop.product.bestselling_product.cache_provider:
```
Comment (vitek-rostislav) [inline project-base/config/packages/cache.yaml:7]: I am thinking about the naming, the "shop" keyword seems like a relic from the ancient times when "ShopBundle" ruled the world :slightly_smiling_face: Few lines above, there is simply `main_friendly_url_slug_cache` so is not it possible to rename it to something like this? E.g. `bestselling_product_cache`? :thinking:

File: upgrade/UPGRADE-v9.2.0-dev.md [hunk 2]
```diff
@@ -27,11 +27,43 @@ There you can find links to upgrade notes for other versions too.
...
+- **\[BC break\]** change entity extension subscriber class ([#2405](https://github.com/shopsys/shopsys/pull/2405))
     - see #project-base-diff to update your project 
     - package joschi127/doctrine-entity-override-bundle is no longer used
     - previously used subscriber `\Joschi127\DoctrineEntityOverrideBundle\EventListener\LoadORMMetadataSubscriber` was replaced with `\Shopsys\FrameworkBundle\Component\EntityExtension\EntityExtensionSubscriber`
         - if you have extended `LoadORMMetadataSubscriber`, you will need to extend `EntityExtensionSubscriber` instead and reimplement your changes on top of the new class
 
 - replace deprecated namespace `Doctrine\Common\Persistence\ObjectManager` with new `Doctrine\Persistence\ObjectManager`([#2407](https://github.com/shopsys/shopsys/pull/2407))
     - see #project-base-diff to update your project
+
+- **\[BC break\]** change used caches from doctrine/cache to Symfony cache ([#2413](https://github.com/shopsys/shopsys/pull/2413))
```
Comment (vitek-rostislav) [inline upgrade/UPGRADE-v9.2.0-dev.md:39]: hmm, I am not able to add a suggestion here :thinking: :disappointed: 

So:
```diff
- [#2413](https://github.com/shopsys/shopsys/pull/2413)
+ [#2412](https://github.com/shopsys/shopsys/pull/2412)
```

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]
Hi, nice job there, I have added just two little notes

File: project-base/config/packages/cache.yaml (hunk 1 above)
Comment (grossmannmartin) [reply project-base/config/packages/cache.yaml]: changed in 8378fe660, thanks!

File: upgrade/UPGRADE-v9.2.0-dev.md (hunk 2 above)
Comment (grossmannmartin) [reply upgrade/UPGRADE-v9.2.0-dev.md]: 👍 
fixed in 202a0913f

**Review by vitek-rostislav — APPROVED** [review]

**Pull request merged**